rooms.js

```diff
@@ -699,6 +699,8 @@ class GlobalRoom extends BasicRoom {
 	 * @param {AnyObject} options
 	 */
 	onCreateBattleRoom(p1, p2, room, options) {
+		if (!room.battle || room.battle.reported) return;
```

Comment:
This should also be inside `addPlayer`, which should also be the only place `onCreateBattleRoom` is called.